halotools/empirical_models/abunmatch/engines/bin_free_cam_kernel.pyx

```diff
@@ -201,58 +227,47 @@ def cython_bin_free_cam_kernel(double[:] y1, double[:] y2, int[:] i2_match, int
     for iy1 in range(nhalfwin, npts1-nhalfwin):
 
         rank1 = correspondence_indx1[nhalfwin]
-        iy2_match = i2_match[iy1]
-
-        #  Stop updating the second window once we reach npts2-nwin/2
-        if iy2_match > iy2_max:
-            iy2_match = iy2_max
+        #  Where to center the second window (making sure we don't slide off the end)
+        iy2_match = min(i2_match[iy1], iy2_max)
 
-        if iy2 > iy2_max:
```

Comment:
I'm pretty sure that this shouldn't be possible and this set of `if if else` block confused me a bit at first. I think this reads a bit easier now but I'm happy to revert if you disagree.

Reply:
Agreed this is a little easier to read, but `min` returns a python object, so it's better practice to use a c-imported min function instead:

`from libc.math cimport fmin as c_min 
`

After looking at the `cython -a` output, in this case I'm pretty sure the cython compiler figures it out anyway and there is no performance hit, so just pointing this out so you are aware of python-cython interactions.

Reply:
:+1: thanks.

Reply:
Actually, as these are `ints` I don't think there is a min/max func in `libc.math` for them. (checked https://github.com/cython/cython/blob/master/Cython/Includes/libc/math.pxd)

http://cython.readthedocs.io/en/latest/src/userguide/numpy_tutorial.html suggests that cython has a fast min/max for ints (as you saw in the `cython -a` output).